### ISSUE TITLE
Fatal Python error: Py_Initialize: Unable to get the locale encoding

### ISSUE THREAD

**Issue by neerajgupta2407** [GrahamDumpleton/mod_wsgi#584]
HI, i have already verified the same from issue no #492 but dispite passing the correct values, i am getting the same error:
```
[Thu Jun 18 12:49:20.708858 2020] [core:notice] [pid 743:tid 140041418251200] AH00051: child pid 812 exit signal Aborted (6), possible coredump in /etc/apache2
Fatal Python error: Py_Initialize: Unable to get the locale encoding
ModuleNotFoundError: No module named 'encodings'

Current thread 0x00007f5deefd1bc0 (most recent call first):
```

My Virtualhost.
```<VirtualHost *:80>
    ServerName api.pricemenu.in

    ErrorLog ${APACHE_LOG_DIR}/error.log
    CustomLog ${APACHE_LOG_DIR}/access.log combined
    Alias /static /home/ubuntu/codes/backend/saleor/static
    <Directory /home/ubuntu/codes/backend/saleor/static>
            FallbackResource disabled
            Require all granted
    </Directory>
    Alias /media /home/ubuntu/codes/backend/saleor/media
    <Directory /home/ubuntu/codes/backend/saleor/media>
            FallbackResource disabled
            Require all granted
    </Directory>
    <Directory /home/ubuntu/codes/backend/saleor/saleor/wsgi>
            FallbackResource disabled
            <Files __init__.py>
                    Require all granted
            </Files>
    </Directory>
    WSGIPassAuthorization on
    WSGIDaemonProcess saleor python-home=/home/ubuntu/codes/backend/venv/  python-path=/home/ubuntu/codes/backend/saleor/
    WSGIProcessGroup saleor
    WSGIScriptAlias / /home/ubuntu/codes/backend/saleor/saleor/wsgi/__init__.py

</VirtualHost>
```

Virtual env Path:
```python -c 'import sys; print(sys.prefix)'```
```Output: /home/ubuntu/codes/backend/venv```

```Code path: /home/ubuntu/codes/backend/saleor```
 Also i have given 777 permission to venv to make sure permission should not be the issue 

Kindly help.
Thanks in Advance.

**Comment by GrahamDumpleton** [comment]
What version/installation of Python is mod_wsgi compiled for?

What version/installation of Python was the Python virtual environment created for?

What are the permissions on the directories:

```
/home/ubuntu
/home/ubuntu/codes
/home/ubuntu/codes/backend
```

It is not just the virtual environment directory that matters.

Also make sure that all directories in tree under the virtual environment root have permissions which would allow access to another user. Do not blindly run ``chmod 0777`` on stuff though as that screws up security of your system.

**Comment by neerajgupta2407** [comment]
@GrahamDumpleton 
What version/installation of Python is mod_wsgi compiled for? `Its not installed for global version as i want to use apache under my venv . My default python version on ubuntu is 2.7. and my venv required version is python 3.7+`

What version/installation of Python was the Python virtual environment created for? `Python 3.7.7 . | mod-wsgi==4.7.1| `

What are the permissions on the directories:
`
/home/ubuntu. :                         drwxr-xr-x 15 ubuntu ubuntu 4096 Jun 18 14:44 ubuntu/
/home/ubuntu/codes.                drwxrwxr-x  3 ubuntu ubuntu  4096 Jun 14 11:53 codes/
/home/ubuntu/codes/backend.  drwxrwxr-x  6 ubuntu ubuntu 4096 Jun 18 14:49 backend/
`

I agreed `chmod 0777` gave full access but in order to make sure permission is not the issue, i gave `0777`

**Comment by GrahamDumpleton** [comment]
Add at global scope (outside of VirtualHost) in your Apache configuration the directive:

```
WSGIRestrictEmbedded ON
```

The issue may be that your Python is in a non standard location, and the error is from the Python interpreters being setup in the Apache child worker process and not mod_wsgi daemon processes.

That directive ensures Python is not setup in Apache child worker processes.

**Comment by neerajgupta2407** [comment]
@GrahamDumpleton  Added `WSGIRestrictEmbedded ON` but still getting same error.

`
[Sun Jun 21 14:29:28.114625 2020] [core:notice] [pid 21544:tid 140584720141248] AH00052: child pid 21613 exit signal Aborted (6)
Fatal Python error: Py_Initialize: Unable to get the locale encoding
ModuleNotFoundError: No module named 'encodings'

Current thread 0x00007fdc6e4e0bc0 (most recent call first):
`

Seems like this issue.
https://stackoverflow.com/questions/51763280/python-virtual-environment-with-python-3-7-and-apache-do-i-have-to-compile-mod

The requirement for Proiect( https://github.com/mirumee/saleor/)  is `python 3.7+`
I am running other Django Project with python 3.6.9 on Ubuntu 18 and they are working perfectly fine.

**Comment by neerajgupta2407** [comment]
@GrahamDumpleton Thanks for your support but i finally figure out the issue.
mod_Wsgi was not complied for python3.7.

Found a nice article about how to build mod_wsgi for python 3.7
Ref: https://medium.com/@garethbjohnson/serve-python-3-7-with-mod-wsgi-on-ubuntu-16-d9c7ab79e03a


I have created  small gist for how to install mod_wsgi for python3.7.
https://gist.github.com/neerajgupta2407/e651c1f815b2f27ba1ff8ba7e507158b

Might be useful to other users..

Thanks
